### PR TITLE
feat(spanner): add instance autoscaling attributed to spanner instance resource

### PR DESCRIPTION
**Pull request by rahul2393**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
Add autoscaler config support for instance resource
```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
Hello, I notice this pr is in draft. Let me know if there is anything I can do to help. As our bot suggests you'll need to add test for these new configurations :)

**Comment by rahul2393** [comment]
@ScottSuarez PR is ready for review

File: mmv1/products/spanner/Instance.yaml [hunk 1]
```diff
@@ -156,3 +156,79 @@ properties:
     values:
       - :READY
       - :CREATING
+  - !ruby/object:Api::Type::NestedObject
+    name: 'autoscalingConfig'
+    description: |
+      The autoscaling configuration. Autoscaling is enabled if this field is set.
+      When autoscaling is enabled, node_count and processing_units are treated as,
```
Comment (ScottSuarez) [inline mmv1/products/spanner/Instance.yaml:163]:
```suggestion
      When autoscaling is enabled, num_nodes and processing_units are treated as,
```

**Review by ScottSuarez — COMMENTED** [review]
nit on description

**Review by ScottSuarez — APPROVED** [review]

**Pull request merged**